src/opencensus/proto/tags/v1/tags.proto

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package opencensus.proto.tags.v1;
+
+option go_package = "github.com/census-instrumentation/opencensus-proto/gen-go/tags/v1";
+
+option java_multiple_files = true;
+option java_package = "io.opencensus.proto.tags.v1";
+option java_outer_classname = "TagsProto";
+
+// TagScope is used to determine the scope of a Tag.
+message TagScope {
+  enum Scope {
+    // Unknown type.
+    TYPE_UNSPECIFIED = 0;
+    // Tag with Local scope are used within the process it created.
+    LOCAL_SCOPE = 1;
+    // A tag is created with the Request scope then it is propagated across
+    // process boundaries subject to outgoing and incoming (on remote side)
+    // filter criteria.
+    REQUEST_SCOPE = 2;
+  }
+
+  // This field is required.
+  Scope scope = 1;
+}
+
+// TagKey is the name of the Tag.
+message TagKey {
+  // Restrictions:
+  //   * MUST contain only printable ASCII (codes between 32 and 126 inclusive)
+  //   * MUST have length greater than zero and less than 256.
+  //   * MUST not be empty.
+  //
+  // This field is required.
+  string value = 1;
+}
+
+// TagValue is the value associated with a Tag.
+message TagValue {
+  // It MUST contain only printable ASCII (codes between 32 and 126)
+  //
+  // This field is required.
+  string value = 1;
+}
+
+// A Tag consists of TagScope, TagKey, and TagValue.
+message Tag {
+  // This field is required.
+  TagKey key = 1;
+
+  // This field is required.
+  TagValue value = 2;
+
+  // This field is optional, if not present the default scope is REQUEST_SCOPE
+  // (the motivation is backwards compatibility).
+  TagScope scope = 3;
+}
+
+// TagMap is an abstract data type that represents collection of tags. i.e.,
+// each key is associated with exactly one value.
+message TagMap {
+  // Not a map becuase the proto3 map does not allow a message as the key.
+  repeated Tag tags = 1;
```

Comment:
so tags with the same name is allowed, correct? Perhaps something should be said about it in comment

Reply:
The list should contains unique TagKeys. The proto3 map does not allow the map_key to be a proto message see https://developers.google.com/protocol-buffers/docs/proto3#maps.

Reply:
The fact that the same key value is not allowed is not clear from the comment at all. As minimum I suggest to indicate this requirement in a comment.

Reply:
BTW, `map` with the `string` key is used for `Attributes` already. Arguably `Attributes` keys should have all the same scenarios as tags.

Reply:
Just to clarify - some misunderstanding may be coming from how you can read the sentence "each key is associated with exactly one value.". If you treat "each key" as unique string - it is already handled. But I read it as one key object maps to one value object. I see you already handled it, just clarifying what I meant,

Reply:
The reason of using a TagKey instead of directly string is because few reasons:
1) Later we can make the TagKey { oneof (string, int) } where int may represent a global id if you have a global registry for all the tags.
2) For the string in TagKey we need to do validation (limited char set) so this way people can create the key once and avoid validation every time in the api.